cpp/src/arrow/compute/exec/sink_node.cc

```diff
@@ -46,31 +46,81 @@ using internal::checked_cast;
 namespace compute {
 namespace {
 
+class BackpressureResevoir : public BackpressureMonitor {
```

Comment:
nit but seems it's spelled Rese**r**voir

Reply:
:exploding_head: Fixed...but this is a serious [Berenstain Bears](https://en.wikipedia.org/wiki/Berenstain_Bears#Name_confusion) moment.